viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -252,6 +256,9 @@ class SLAMServiceImpl final : public SLAMService::Service {
     // started.
     std::string latest_pointcloud_map;
     // ---
+
+    cartographer::mapping::TrajectoryBuilderInterface *trajectory_builder;
```

Comment:
1. Why does this need to be global?
2. Where is this pointer initialized?
3. Is it possible for `AddSensorReading` to be called before `ProcessDataAndStartSavingMaps` is called & causes memory corruption due to this pointer pointing at invalid memory?
4. When do we free the object this pointer points to?
5. If we really need a pointer on an object, could we construct this at CPU-bound initialization time (that way we are guaranteed it will always be a valid pointer) & have this be a smart pointer?

Reply:
1. I believe we put this definition here so that it could be accessed within `AddSensorReading` so that we did not have to pass it in and change the function signature
2. I think it is initialized [here](https://github.com/viamrobotics/viam-cartographer/blob/93f5c18e56da635fb74a9d6f8896bb1815f7a9e0/viam-cartographer/src/slam_service/slam_service.cc#L657) within `map_builder.SetTrajectoryBuilder()`
3. I think that would be a problem yes, we talked about moving it into the initialization function @kkufieta is working on as a separate PR
4. I think that would happen in the terminate function [this ticket](https://viam.atlassian.net/browse/RSDK-3327) refers to
5. I am a little confused on what you mean here - would putting it in Kats init function be what you are talking aout?